src/Compilers/CSharp/Portable/CommandLine/CSharpCompiler.cs

```diff
@@ -89,7 +89,7 @@ protected CSharpCompiler(CSharpCommandLineParser parser, string? responseFile, s
             }
 
             // If errors had been reported in ParseFile, while trying to read files, then we should simply exit.
-            if (ReportDiagnostics(diagnosticBag.ToReadOnlyAndFree(), consoleOutput, errorLogger))
+            if (ReportDiagnostics(diagnosticBag.ToReadOnlyAndFree(), consoleOutput, errorLogger, null))
```

Comment:
Consider using named arguments for these parameters. There are a number of these throughout this PR.